src/chat/narrowsReducer.js

```diff
@@ -189,6 +231,33 @@ export default (
     case EVENT_MESSAGE_DELETE:
       return eventMessageDelete(state, action);
 
+    case EVENT_UPDATE_MESSAGE: {
+      // Compare the corresponding caughtUpReducer case.
+
+      let result: NarrowsState = state;
+      const { event, move } = action;
+
+      if (move) {
+        // The edit changed topic and/or stream.
+        const { orig_stream_id, orig_topic, new_stream_id, new_topic } = move;
+        const messageIdSet = new Set(event.message_ids);
+        result = addMessages(result, topicNarrow(new_stream_id, new_topic), event.message_ids);
+        result = removeMessages(result, topicNarrow(orig_stream_id, orig_topic), messageIdSet);
+        if (new_stream_id !== orig_stream_id) {
+          result = addMessages(result, streamNarrow(new_stream_id), event.message_ids);
```

Comment:
When adding the messages, we'll want them in sorted order. Partly because that's the order in which we want them in the result, and partly because it lets us quickly look at the first and last one to answer questions like "are these all contained in the interval where we know we have all this narrow's messages".